xblocks_contrib/annotatable/annotatable.py

```diff
@@ -0,0 +1,115 @@
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    # TO-DO: change this view to display your data your own way.
+    def student_view(self, context=None):
+        """
+        Create primary view of the AnnotatableXBlock, shown to students when viewing courses.
+        """
+        if context:
+            pass  # TO-DO: do something based on the context.
+        html = self.resource_string("static/html/annotatable.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/annotatable.css"))
+
+        # Add i18n js
+        statici18n_js_url = self._get_statici18n_js_url()
+        if statici18n_js_url:
+            frag.add_javascript_url(
+                self.runtime.local_resource_url(self, statici18n_js_url)
+            )
+
+        frag.add_javascript(self.resource_string("static/js/src/annotatable.js"))
+        frag.initialize_js("AnnotatableXBlock")
+        return frag
+
+    # TO-DO: change this handler to perform your own actions.  You may need more
+    # than one handler, or you may not need any handlers at all.
+    @XBlock.json_handler
+    def increment_count(self, data, suffix=""):
+        """
+        Increments data. An example handler.
+        """
+        if suffix:
+            pass  # TO-DO: Use the suffix when storing data.
+        # Just to show data coming in...
+        assert data["hello"] == "world"
+
+        self.count += 1
+        return {"count": self.count}
+
+    # TO-DO: change this to create the scenarios you'd like to see in the
+    # workbench while developing your XBlock.
+    @staticmethod
+    def workbench_scenarios():
+        """Create canned scenario for display in the workbench."""
+        return [
+            (
+                "AnnotatableXBlock",
+                """<annotatable/>
+             """,
+            ),
+            (
+                "Multiple AnnotatableXBlock",
+                """<vertical_demo>
+                <annotatable/>
+                <annotatable/>
+                <annotatable/>
+                </vertical_demo>
+             """,
+            ),
+        ]
+
+    @staticmethod
+    def _get_statici18n_js_url():
+        """
+        Return the Javascript translation file for the currently selected language, if any.
+
+        Defaults to English if available.
+        """
+        locale_code = translation.get_language()
+        if locale_code is None:
+            return None
+        text_js = "public/js/translations/{locale_code}/text.js"
+        lang_code = locale_code.split("-")[0]
+        for code in (locale_code, lang_code, "en"):
+            loader = ResourceLoader(__name__)
+            if pkg_resources.resource_exists(
+                loader.module_name, text_js.format(locale_code=code)
+            ):
+                return text_js.format(locale_code=code)
+        return None
+
+    @staticmethod
+    def get_dummy():
```

Comment:
Adding as a reminder, we will remove it in the real implementation of XBlock as it's just a testing method .